core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/UpgradePermissionProvider.java

```diff
@@ -22,12 +23,16 @@
     @Inject
     private ComponentVersionComparator componentVersionComparator;
 
+    @Inject
+    private SupportedRuntimes supportedRuntimes;
+
     public boolean permitCmUpgrade(ImageFilterParams imageFilterParams, Image image) {
         return permitUpgrade(imageFilterParams, image, ImagePackageVersion.CM, ImagePackageVersion.CM_BUILD_NUMBER, false);
     }
 
     public boolean permitStackUpgrade(ImageFilterParams imageFilterParams, Image image) {
-        return permitUpgrade(imageFilterParams, image, ImagePackageVersion.STACK, ImagePackageVersion.CDH_BUILD_NUMBER, checkUpgradeMatrix(imageFilterParams));
+        return permitUpgrade(imageFilterParams, image, ImagePackageVersion.STACK, ImagePackageVersion.CDH_BUILD_NUMBER, checkUpgradeMatrix(imageFilterParams))
+                && isRuntimeVersionSupported(image);
```

Comment:
Can you please flip the two parts of the `AND` statement to run the `isRuntimeVersionSupported ` first?

Reply:
done